src/content/operations/cleanup/_index.en.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Uninstalling Submariner"
+date: 2020-12-23T21:25:11+01:00
+weight: 50
+---
+
+To properly uninstall Submariner from a cluster, follow the steps below:
+
+{{% notice note %}}
+Make sure KUBECONFIG for all participating clusters is exported and all participating clusters are accessible via kubectl.
+{{% /notice %}}
+
+1. Delete Submariner-related namespaces
+
+   For each participating cluster, issue the following command:
+
+   ```bash
+   kubectl delete namespace submariner-operator
+   ```
+
+   For the Broker cluster, issue the following command:
+
+   ```bash
+   kubectl delete namespace submariner-k8s-broker
+    ```
+
+2. Delete the Submariner CRDs
+
+   For each participating cluster, issue the following command:
+
+   ```bash
+   for CRD in `kubectl get crds | grep -iE 'submariner|multicluster.x-k8s.io'| awk '{print $1}'`; do kubectl delete crd $CRD; done
+   ```
+
+3. Delete Submariner's `ClusterRole`s and `ClusterRoleBinding`s
+
+   For each participating cluster, issue the following command:
+
+   ```bash
+   roles="submariner-operator submariner-operator-globalnet submariner-lighthouse submariner-networkplugin-syncer"
+   kubectl delete clusterrole,clusterrolebinding $roles --ignore-not-found
+   ```
+
+4. Remove the Submariner gateway labels
+
+   For each participating cluster, issue the following command:
+
+   ```bash
+   kubectl label --all node submariner.io/gateway-
+   ```
+
+5. For OpenShift deployments, delete Lighthouse entry from `default` DNS.
+
+   For each participating cluster, issue the following command:
+
+   ```bash
+   kubectl apply -f -  <<EOF
+   apiVersion: operator.openshift.io/v1
+   kind: DNS
+   metadata:
+     finalizers:
+     - dns.operator.openshift.io/dns-controller
+     name: default
+   spec:
+     servers: []
+   EOF
+   ```
+
+6. Delete Submariner-related configmaps
+
+   For each participating cluster, issue the following commands:
+
+   ```bash
+   kubectl delete configmap submariner-engine-lock -n submariner-operator
+   kubectl delete configmap submariner-operator-lock -n submariner-operator
+   kubectl delete configmap submariner-lighthouse-coredns -n submariner-operator
+   ```
```

Comment:
In Step 1, if we delete the `submariner-operator` namespace, all the configmaps will be deleted automatically.

Reply:
Correct, this step 6 is not needed after the namespace is gone:

```
$ kubectl delete configmap submariner-engine-lock -n submariner-operator
Error from server (NotFound): configmaps "submariner-engine-lock" not found
$ kubectl delete configmap submariner-operator-lock -n submariner-operator
Error from server (NotFound): configmaps "submariner-operator-lock" not found
$ kubectl delete configmap submariner-lighthouse-coredns -n submariner-operator
Error from server (NotFound): configmaps "submariner-lighthouse-coredns" not found
```